cyral/resource.go

```diff
@@ -57,7 +58,7 @@ func CreateResource(createConfig ResourceOperationConfig, readConfig ResourceOpe
 		}
 		log.Printf("[DEBUG] Response body (unmarshalled): %#v", createConfig.ResponseData)
 
-		createConfig.ResponseData.WriteToSchema(d)
+		createConfig.ResourceData.WriteToSchema(d)
```

Comment:
The return `ReadResource` already uses the `ResponseData` to write to the schema, which makes this call redundant.

Reply:
Not really, depending on the struct type that is being passed and the different implementations of the `WriteToSchema` method, this could end up changing the resource behavior. For instance, consider the `resource_cyral_integration_datadog`, it calls [this WriteToSchema](https://github.com/cyralinc/terraform-provider-cyral/blob/8d97b8ee2c48cde216be794d405c95903d7711d3/cyral/resource.go#L33) in the `CreateResource`, but in the `ReadResource` a [different WriteToSchema](https://github.com/cyralinc/terraform-provider-cyral/blob/main/cyral/resource_cyral_integration_datadog.go#L17) is called, this is also valid for other resources. Btw, I agree that the `resource.go` needs a general refactoring, but ideally, this needs to be done in a separate and specific task, since a lot of resources depend on this file.